client/src/components/CommentApp/comments.js

```diff
@@ -66,8 +102,15 @@ class BasicFieldLevelAnnotation {
   hide() {
     this.node.classList.add('u-hidden');
   }
-  setOnClickHandler(handler) {
-    this.node.addEventListener('click', handler);
+  setOnClickHandler(localId) {
+    this.node.addEventListener('click', () => {
+      this.commentApp.store.dispatch(
+        this.commentApp.actions.setFocusedComment(localId)
+      );
+      this.commentApp.store.dispatch(
+        this.commentApp.actions.setPinnedComment(localId)
+      );
```

Comment:
I can see that in two cases those two actions are used at the same time. If that’s always the case, that should be a single action to not lead to potentially inconsistent state.

Reply:
This isn't always the case - they're used separately in the commenting system itself. For widgets, it usually makes sense to do both together, so I could combine them in one utility action if you prefer for their use, but it doesn't lead to inconsistent state if done separately.